docs/service-fabric/refactor-migrated-app.md

```diff
@@ -0,0 +1,155 @@
+
+**Tailspin.SurveyResponseService** is a stateful service that stores survey answers in Azure Blob Storage. It also merges answers into the survey analysis data. The service is implemented as a stateful service because it uses a reliable queue to process survey answers in batches.
+
+**Tailspin.SurveyManagementService** is a stateless service that stores and retrieves surveys and survey questions. The service uses Azure Blob storage.
+
+**Tailspin.SurveyAnswerService** is a stateless service that retrieves survey answers and survey analysis. The service also uses Azure Blob storage.
+
+**Tailspin.SurveyAnalysisService** is a stateless service that persists survey answer summary data in a Redis cache. This service is called by the *Tailspin.SurveyResponseService* each time a survey is answered and the new survey answer data is merged in the summary data. In the original Cloud Services application, this functionality was implemented in the *Tailspin.Workers.Survey* role.
+
+## Stateless versus stateful services
+
+Azure Service Fabric supports three programming models:
+* The Guest Executable model allows any executable to be packaged as a service and deployed to a Service Fabric cluster. Service Fabric orchestrates and manages execution of the guest executable.
+* The Reliable Services programming model allows for the creation of stateless or stateful services that integrate with all Service Fabric platform features. Stateful services allow for replicated state to be stored in the Service Fabric cluster. Stateless services do not.
+* The Reliable Actors programming model allows for the creation of services that implement the virtual actor pattern.
+
+All the services in the Surveys application are stateless reliable services, except for the *Tailspin.SurveyResponseService* service. This service implements a [ReliableConcurrentQueue][reliable-concurrent-queue] to process survey answers when they are received. Responses placed in the ReliableConcurrentQueue are saved into Azure Blob Storage and passed to the *Tailspin.SurveyAnalysisService* for analysis.
+
+Note that operations to persist dequeued items from a ReliableConcurrentQueue should ideally be idempotent. If an exception is thrown during the processing of an item from the queue, the same item may be processed more than once. In the Surveys application, the operation to merge survey answers to the *Tailspin.SurveyAnalysisService* is not idempotent because Tailspin decided that the survey analysis data is only a current snapshot of the analysis data and does not need to be consistent. The survey answers saved to Azure Blob Storage are eventually consistent, so the survey final analysis can always be recalculated correctly from this data.
```

Comment:
> does not need to be consistent ... so the survey final analysis can always be recalculated correctly from this data.

Is recalculating a manual process? If there are errors, will the results continue to drift over time until corrected?

Reply:
Currently yes. We could implement a batch update (Lambda Architecture) to fix errors in the "fast" processing of the Analysis data, but not in current codebase.